quartz/components/scripts/search.inline.ts

```diff
@@ -110,21 +133,67 @@ document.addEventListener("nav", async (e: unknown) => {
     }
   }
 
+  function trimContent(content: string) {
+    // TODO: could be optimized by adding heuristic to substring before splitting for better performance with big files
+    // Split words and splice at max words
+    const words = content.split(" ")
+    if (words.length > contextWindowWords) {
+      words.splice(contextWindowWords)
+
+      // Add trailing ... to indicate cutoff
+      words[words.length - 1] += "..."
+    }
+    return words.join(" ")
+  }
+
   const formatForDisplay = (term: string, id: number) => {
     const slug = idDataMap[id]
     return {
       id,
       slug,
       title: highlight(term, data[slug].title ?? ""),
-      content: highlight(term, data[slug].content ?? "", true),
+      // if searchType is tag, display context from start of file and trim, otherwise use regular highlight
+      content:
+        searchType === "tags"
+          ? trimContent(data[slug].content)
+          : highlight(term, data[slug].content ?? "", true),
+      tags: highlightTags(term, data[slug].tags),
```

Comment:
```suggestion
      tags: highlightTags(searchType === "tags" ? term : "", data[slug].tags),
```

probably don't highlight tags for non tag search

Reply:
the first line in `highlightTags` already checks for searchType and returns `[]` if its not `tags`. The check could be extracted out of `higlightTags` into something like `tags: searchType === "tags" ? highlighTags(...) : []`, but I dont think we need the check twice (both in function and outside).